add register and login requirements

Mike5801 opened the pull request:
Agregará toda la lógica para registrarse como usuario con un token de celular y del mismo modo iniciar sesión. También incluye refactorización del código para algunas vistas y la obtención de la información del usuario que intenta iniciar sesión o en dado caso registrarse. También se añadio dos radio buttons correspondientes al género y otro para seleccionar el rol que quiere fungir en la aplicación. Incluye del mismo modo toda la implementación con firebase de consultas desde el registro del usuario a éste y de sus roles y de su inicio de sesión y obtención de su información. Actualmente si corren la aplicación la primer pantalla estará para visualizar la pantalla para poder iniciar sesión

Mike5801 committed:
change login and register activity

Renato549 committed:
change login and register activity (#16)

Change constraint layouts in register activity and in login activity

Renato549 committed:
add  OTP and phone Activity

Mike5801 committed:
add  OTP and phone Activity (#17)

Mike5801 committed:
refactor phone authentication, sms validation and register activity code

Commit:
refactor phone authentication, sms validation and register activity code (#18)

Mike5801 committed:
add MVVM clean architecture files and logic for Register User

Mike5801 committed:
add user validation in authentication

Mike5801 committed:
add get user data when authenticating

Mike5801 committed:
change register to radio buttons and fix edit text on otpactivity

Mike5801 committed:
add get role and assing role to user requirement

Mike5801 committed:
refactor adding user and role function

Mike5801 committed:
add get role activity

Mike5801 committed:
add CheckIfLogged activity to verify if logged in

Commit:
add register and log in requirements (#19)

Agregará toda la lógica para registrarse como usuario con un token de
celular y del mismo modo iniciar sesión. También incluye refactorización
del código para algunas vistas y la obtención de la información del
usuario que intenta iniciar sesión o en dado caso registrarse. También
se añadio dos radio buttons correspondientes al género y otro para
seleccionar el rol que quiere fungir en la aplicación

Commit:
Merge branch 'Equipo2' into Equipo2

Mike5801 committed:
Merge pull request #26 from MiguelWTangF/Equipo2

add register and login requirements

Mike5801 committed:
Merge branch 'develop' into Equipo2

MandukoGR commented on app/src/main/AndroidManifest.xml:
```diff
@@ -28,10 +28,6 @@
             android:name=".framework.view.PhoneActivity"
             android:exported="true">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
```
aqui va a tronar, tienes dos actividades con intent-filter

MandukoGR commented on app/src/main/AndroidManifest.xml:
```diff
@@ -44,7 +40,14 @@
             android:exported="true">
         </activity>
-
+        <activity
```
Este es el otro activity con intent-filter

Mike5801 replied:
Creo que solo hay un intent-filter, el otro aparece en rojo osea es código que quitamos, o bueno no me aparece el segundo

MandukoGR commented on app/src/main/java/com/example/appatemporal/framework/view/OTPActivity.kt:
```diff
@@ -168,21 +155,19 @@ class OTPActivity : AppCompatActivity() {
-        inputOTP5 = findViewById(R.id.otpEditText5)
-        inputOTP6 = findViewById(R.id.otpEditText6)
+        progressBar = binding.otpProgressBar
+        verifyBtn = binding.verifyOTPBtn
```
Es necesario guardarlo en variables? El binding es justo para evitar tanto código

MandukoGR commented on app/src/main/java/com/example/appatemporal/framework/view/OTPActivity.kt:
```diff
@@ -195,21 +180,16 @@ class OTPActivity : AppCompatActivity() {
-
-
+                binding.otpEditText1.id -> if(text.length == 1 ) inputOTP2.requestFocus() else if (text.isEmpty()) inputOTP1.requestFocus()
+                binding.otpEditText2.id -> if(text.length == 1 ) inputOTP3.requestFocus() else if (text.isEmpty()) inputOTP1.requestFocus()
```
De hecho creo que no están ocupando las variables definidas anteriormente, vuelven a llamar desde el binding

MandukoGR reviewed (commented):
Checa los comentarios que puse, todo lo demás lo veo bien

Mike5801 committed:
refactor OTPActivity code using view binding

Commit:
refactor OTPActivity code using view binding (#20)

Cambia el uso de variables para linkear los elementos de las vistas por
viewBinding

Mike5801 committed:
Merge pull request #28 from MiguelWTangF/Equipo2

refactor OTPActivity code using viewBinding
